src/distro/distro.py

```diff
@@ -1078,6 +1082,47 @@ def uname_attr(self, attribute: str) -> str:
         """
         return self._uname_info.get(attribute, "")
 
+    def __resolve_chroot_symlink(self, link_location: str) -> str:
+        # only resolve symlink if root_dir is set and link is under it
+        if (
+            self.root_dir is None
+            or os.path.commonprefix([self.root_dir, link_location]) != self.root_dir
+        ):
+            return link_location
+
+        seen_paths = []
+        while True:
+            try:
+                resolved = os.readlink(link_location)
+            except OSError:  # includes case "not a symlink"
+                return link_location
+
+            if os.path.isabs(resolved):
+                # i.e. absolute path (regarding to the chroot), that we need to
+                # "move" back inside the chroot
+                resolved = os.path.splitdrive(resolved)[1].lstrip(os.sep)
+                if os.altsep is not None:
+                    resolved.lstrip(os.altsep)
+                resolved = os.path.join(self.root_dir, resolved)
+            else:
+                # i.e. relative path that we make absolute
+                resolved = os.path.join(os.path.dirname(link_location), resolved)
+
+            # remove redundant up-level references
+            resolved = os.path.normpath(resolved)
+
+            # prevent symlink infinite loop
+            if resolved in seen_paths:
+                raise OSError(errno.ELOOP, os.strerror(errno.ELOOP), resolved)
+
+            if os.path.commonprefix([self.root_dir, resolved]) != self.root_dir:
+                raise RuntimeError(
+                    f"{link_location} resolves outside of {self.root_dir}!"
+                )
```

Comment:
I'm not sure `RuntimeError` is the best exception to throw. I think we have another flavor of `OSError` here, maybe even `FileNotFoundError`.
PS: My vote for dropping the exclamation mark. If we add one here, all other exceptions could be argued to need one (but probably don't, either).

Reply:
You are right, I rolled-back to `PermissionError` (which seems to fit better), and redacted the `!`.